config/routes.rb

```diff
@@ -63,20 +63,63 @@
   ##################################################
   devise_for :students, controllers: {
     sessions: 'authentication/students/sessions',
-    registrations: 'authentication/students/registrations'
+    registrations: 'authentication/students/registrations',
+    passwords: 'authentication/students/passwords'
   }
 
   devise_for :teachers, controllers: {
     sessions: 'authentication/teachers/sessions',
-    registrations: 'authentication/teachers/registrations'
+    registrations: 'authentication/teachers/registrations',
+    passwords: 'authentication/teachers/passwords'
+
   }
 
   devise_for :admins, controllers: {
     sessions: 'authentication/admins/sessions',
-    registrations: 'authentication/admins/registrations'
+    registrations: 'authentication/admins/registrations',
+    passwords: 'authentication/admins/passwords'
   }
 
   ##################################################
+  #
+  # Change Passwords
+  #
+  ##################################################
+
+  devise_scope :student do
+    post 'passwords/students/reset_request', to: 'authentication/students/passwords#send_token'
+    post 'passwords/students/reset_password', to: 'authentication/students/passwords#reset_password'
+  end
+
+  resource :student, only: [:edit] do
```

Comment:
It turns out we actually do! I think the patch is more of a middleman between the erb file and the actual controller function. When I tried removing this post request, I end up getting:

![image](https://cloud.githubusercontent.com/assets/7674590/23376151/fb56a7e6-fcdf-11e6-8bc2-0ecdbf7af109.png)